src/Network/Bugsnag/Exception.hs

```diff
@@ -73,12 +73,15 @@ bugsnagException errorClass message stacktrace = BugsnagException
 -- ("IOException",Just "user error (Oops)")
 --
 bugsnagExceptionFromSomeException :: SomeException -> BugsnagException
-bugsnagExceptionFromSomeException ex =
-    foldr go (bugsnagExceptionWithParser parseStringException ex) exCasters
-  where
+bugsnagExceptionFromSomeException ex = foldr go seed exCasters
+ where
     go :: Caster -> BugsnagException -> BugsnagException
     go (Caster caster) res = maybe res caster $ fromException ex
 
+    seed = (bugsnagExceptionWithParser parseStringException ex)
+        { beErrorClass = (\(SomeException e) -> exErrorClass e) ex
```

Comment:
I'm not sure if `exCasters` are still necessary with this added. There is some variation in the exception constructor for each caster `bugsnagExceptionWithParser` vs `bugsnagExceptionFromException`.

Reply:
That'd be great if we can simplify that out, I think I'd have see an example of our how existing usage changes to (e.g.) set group-by from request domain in `HttpException`s. Ideally, we'd back-fill some test cases around this.